src/crypto/asset_proofs.cpp

```
@@ -0,0 +1,952 @@
+
+    if (!verify_BGE_proof(context_hash, ring, T, out))
+        return false;
+
+    return true;
+}
+
+// ---------------------------------------------------------------------------
+// BGE verify
+// ---------------------------------------------------------------------------
+
+bool verify_BGE_proof(const rct::key&    context_hash,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)